lib/src/pubspec.dart

```diff
@@ -409,6 +420,25 @@ class Pubspec extends PubspecBase {
     _collectError(_ensureEnvironment);
     return errors;
   }
+}
+
+/// A mixin that contains the shared logic for parsing of [Pubspec] and
+/// [PubspecOverrides].
+mixin _PubspecParsingMixin {
```

Comment:
Would it be possible to just use top-level functions for doing the dependency-map parsing? I feel that a mixin is overly complicating things here.

Something like:

```
Map<String, PackageRange> parseDependencyMap(String field, YamlNode? node, LanguageVersion laguageVersion, SourceRegistry sources) {...}

Reply:
`_parseDependencies` uses a few smaller utilitly methods. Some of those use the fileds of `_PubspecParsingMixin` (or before `Pubspec`). They would have to become top-level functions, too, and they would need a few more arguments.

`Pubspec._parseEnvironment` and `Pubspec._computeFeatures` use those utilitly methods aswell.

How about use composition by pulling the parsing logic out into a `_PubspecParser` to avoid the mixin?